pkg/arvo/sys/vane/gall.hoon

```diff
@@ -1067,26 +1067,36 @@
       |=  =ship
       ^+  ap-core
       =/  in=(list [=duct =^ship =path])
-        ~(tap by incoming.subscribers.current-agent)
-      |-  ^+  ap-core
-      ?^  in
-        =?  ap-core  =(ship ship.i.in)
-          =/  core  ap-load-delete(agent-duct duct.i.in)
+        %+  skim
+          ~(tap by incoming.subscribers.current-agent)
+        |=([duct who=^ship path] =(who ship))
+      =/  out=(list [[=wire =^ship =term] ? =path])
+        %+  skim
+          ~(tap by outgoing.subscribers.current-agent)
+        |=([[wire who=^ship term] ? path] =(who ship))
+      ::
+      =.  ap-core
+        %+  roll  in
+        |=  [[=duct *] _ap-core]
+        =/  core  ap-load-delete(agent-duct duct)
+        core(agent-duct agent-duct)
+      =.  ap-core
+        %+  roll  out
+        |=  [[[=wire =^ship =term] ? path] _ap-core]
+          =/  core
+            =.  agent-duct  system-duct.agents.state
+            =/  way  [%out (scot %p ship) term wire]
+            (ap-specific-take way %kick ~)
           core(agent-duct agent-duct)
-        $(in t.in)
       ::
-      =/  out=(list [[=wire =^ship =term] ? =path])
-        ~(tap by outgoing.subscribers.current-agent)
-      |-  ^+  ap-core
-      ?~  out
+      ?:  &(?=(~ in) ?=(~ out))
```

Comment:
Shouldn't `+on-rift` be called regardless of the amount of incoming and outgoing subscriptions?

Reply:
Which ships should you receive notifications about?  Gall hears about breaches for any ships it's talked to, but I don't think an app should receive notifications about a ship it hasn't talked to.